### PR TITLE
[WIP] Attempt to fix travis timeouts

### PR DESCRIPTION
**Pull request by jchodera**
Fiddling with a fix for #1315.

### CODE REVIEW

**Comment by rmcgibbo** [comment]
Might also help to add `--output-on-failure` (https://cmake.org/cmake/help/v2.8.8/ctest.html#opt:--output-on-failure)

File: .travis.yml [hunk 1]
```diff
@@ -145,7 +145,8 @@ script:
     fi
 
   - # run all of the tests, making sure failures at this stage don't cause travis failures
-  - python devtools/run-ctest.py -j2 --schedule-random --stop-time $CTEST_STOP_TIME
+  - # set a timeout of $CTEST_TIMEOUT seconds for each individual test
+  - python devtools/run-ctest.py -j2 --schedule-random --stop-time $CTEST_STOP_TIME --timeout $CTEST_TIMEOUT
   - if [[ ! -z "${DOCS_DEPLOY}" && "${DOCS_DEPLOY}" = "true" ]]; then
```
Comment (rmcgibbo) [inline .travis.yml]: It doesn't look like like the environment variable CTEST_TIMEOUT was ever set though?

Comment (jchodera) [inline .travis.yml]: Whoops!

**Comment by jchodera** [comment]
Either `ctest` is ignoring the command-line arguments, or it isn't actually being passed them.

**Comment by rmcgibbo** [comment]
There could be something wrong with devtools/run-ctest.py, but I looked and didn't see anything.

**Comment by jchodera** [comment]
Well, I can only conclude that `ctest` is completely broken, since it doesn't seem to be honoring any of its command-line arguments.

**Comment by rmcgibbo** [comment]
Or `devtools/run-ctest.py`?

**Comment by jchodera** [comment]
The argument array is definitely being built. I have it being output to the terminal.

**Comment by jchodera** [comment]
Oh.  I think `subprocess.call` should *either* be getting an argument list *or* be getting a single command along with the `shell=True` argument, but not both.

See http://stackoverflow.com/questions/16645761/subprocess-call-doesnt-work-as-expected

**Comment by jchodera** [comment]
That was it.  Hopefully can finish this up soon.

**Comment by rmcgibbo** [comment]
I think the second invocation of ctest from the python script is supposed to not receive the `--stop-time` argument.

**Comment by jchodera** [comment]
Well, what's our goal here?  Do we want to finish whatever tests we can by the time travis allots us runs out, or do we want to run into the travis limit by letting those test reruns run without limits?

**Comment by rmcgibbo** [comment]
We'd like to finish before the job times out. If you pass the same 25 minute timeout to both invocations, then the 1st invocation will get killed after running for 25 minutes and the second invocation will die immediately, because 25 minutes will already have elapsed.

**Comment by rmcgibbo** [comment]
Let me just rewrite it.

**Comment by rmcgibbo** [comment]
Is `-j4` really the best here?

**Comment by jchodera** [comment]
> We'd like to finish before the job times out. If you pass the same 25 minute timeout to both invocations, then the 1st invocation will get killed after running for 25 minutes and the second invocation will die immediately, because 25 minutes will already have elapsed.

Yes, but how do we apportion the available runtime?  Should we timeout individual tests more quickly?

Really, we should probably set the max test run time at the very beginning when travis starts and make sure travis won't time out on us.  Then we could just complete as many tests as we have time for in that time, keeping the per-test timeout to be pretty short.

**Comment by jchodera** [comment]
> Let me just rewrite it.

Go for it!

> Is -j4 really the best here?

I was just trying other things.  It says there is ~2 cores available for this class of VM.

Compilation only takes ~3 min, though.

**Comment by jchodera** [comment]
More importantly, we only want to rerun tests that did not time out.  Is there an easy way to do that?

**Comment by rmcgibbo** [comment]
Oh, good idea. Not sure. I think ctest can generate xUnit output, which might contain it...

**Comment by jchodera** [comment]
It definitely prints this information to the terminal:
```
The following tests FAILED:
	 71 - TestOpenCLEwaldMixed (Timeout)
	 72 - TestOpenCLEwaldDouble (Timeout)
	 77 - TestOpenCLCustomManyParticleForceMixed (SEGFAULT)
	 78 - TestOpenCLCustomManyParticleForceDouble (SEGFAULT)
	100 - TestOpenCLMonteCarloBarostatSingle (Timeout)
	101 - TestOpenCLMonteCarloBarostatMixed (Timeout)
	117 - TestOpenCLRpmdDouble (Timeout)
	118 - TestOpenCLDrudeSCFIntegratorSingle (Timeout)
	119 - TestOpenCLDrudeSCFIntegratorMixed (Timeout)
	124 - TestOpenCLDrudeLangevinIntegratorSingle (Timeout)
	125 - TestOpenCLDrudeLangevinIntegratorMixed (Timeout)
	126 - TestOpenCLDrudeLangevinIntegratorDouble (Timeout)
```

**Comment by rmcgibbo** [comment]
http://stackoverflow.com/a/21688776/1079728 says that you can get xml output.

**Comment by jchodera** [comment]
Closing this in favor of #1317

**Pull request closed without merging**